.github/workflows/test_on_push.yml

```
@@ -97,35 +97,36 @@ jobs:
         run: |
           python -m pip install --upgrade pip wheel setuptools
           python -m pip install "tox<4"
+          python -m pip install "nox"
 
       - name: Install SuiteSparse and Sundials
         if: matrix.os == 'ubuntu-latest'
-        run: tox -e pybamm-requires
+        run: nox -s pybamm-requires
 
-      - name: Run unit tests for GNU/Linux with Python 3.8, 3.9, and 3.10
-        if: matrix.os == 'ubuntu-latest' && matrix.python-version != 3.11
-        run: python -m tox -e unit
+      - name: Run unit tests for GNU/Linux with Python 3.8, 3.10, and 3.11
+        if: matrix.os == 'ubuntu-latest' && matrix.python-version != 3.9
+        run: nox -s unit
```

```suggestion
      - name: Run unit tests for GNU/Linux with Python 3.8, 3.9, and 3.10
        if: matrix.os == 'ubuntu-latest' && matrix.python-version != 3.11
        run: nox -s unit
```

Maybe this was reversed when merging changes from #3016 into your branch

---

Ditto